### ISSUE TITLE
Unable to set volumeClaim labels using variables

### ISSUE THREAD

**Issue by withnale** [argoproj/argo-workflows#6725]
## Summary 

It's not possible at present to set volumeClaim labels using variables. This is possible with pod labels for each of the steps. This breaks our current cost reporting, which expects to have a common set of labels for each kubernetes resource.

For example...

If you take the example workflow `examples/ci.yaml` and modify it as below:

```yaml
  volumeClaimTemplates:
  - metadata:
      name: workdir
      labels:
        foo: "{{workflow.parameters.revision}}"
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 1Gi
```

... you will see the following message when you submit it.

```
Name:                ci-example-rp2zw
Namespace:           main
ServiceAccount:      default
Status:              Error
Message:             pvc create error: PersistentVolumeClaim "ci-example-rp2zw-workdir" is invalid: metadata.labels: Invalid value: "{{workflow.parameters.revision}}": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
```

## Diagnostics

👀 Yes! We need all of your diagnostics, please make sure you add it all, otherwise we'll go around in circles asking you for it:

What Kubernetes provider are you using? GKE v1.20

What version of Argo Workflows are you running? v3.1.2

What executor are you running? Kubelet

Did this work in a previous version? I.e. is it a regression? No

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
The place the code needs to change is here:

https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/operator.go#L1432

Would you like to submit PR to fix?

**Comment by denis-codefresh** [comment]
I was going to to submit a PR but I wasn't able to reproduce it with Argo v3.1.2. For me it looks like this have been fixed with this PR: #5565. And v3.1.2 has this changes.
